### PR TITLE
bugfix: ZENKO-1195 Object monitoring during retry

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Depends on https://github.com/scality/Arsenal/pull/576.

During a retry any object-level metrics keys should be deleted so the new operation's progress is tracked and is not affected by the previous operation.

### CODE REVIEW

**Comment by bennettbuchanan** [comment]
@bert-e wait

**Comment by bennettbuchanan** [comment]
@bert-e reset

File: lib/api/BackbeatAPI.js [hunk 1]
```diff
@@ -601,33 +605,29 @@ class BackbeatAPI {
...
+        const keys = [
+            getObjectBytesKey(storageClass, bucket, key, versionId),
+            getObjectBytesDoneKey(storageClass, bucket, key, versionId),
+        ];
+        return async.each(keys, (key, next) =>
+            this._redisClient.keys(`${key}*`, (err, res) => {
+                if (err) {
+                    return next(err);
+                }
+                return async.eachLimit(res, 10, (key, done) =>
```
Comment (alexanderchan-scality) [inline lib/api/BackbeatAPI.js:626]: the implementation looks good. was wondering if it is possible to use `batch` for deleting the multiple keys

**Review by alexanderchan-scality — APPROVED** [review]

File: lib/api/BackbeatAPI.js [hunk 2]
```diff
@@ -601,33 +605,29 @@ class BackbeatAPI {
...
-            }
-            return cb();
-        });
+    _deleteProgressKeys(bucket, key, versionId, storageClass, cb) {
+        const keys = [
+            getObjectBytesKey(storageClass, bucket, key, versionId),
+            getObjectBytesDoneKey(storageClass, bucket, key, versionId),
+        ];
+        return async.each(keys, (key, next) =>
+            this._redisClient.keys(`${key}*`, (err, res) => {
```
Comment (jonathan-gramain) [inline lib/api/BackbeatAPI.js:622]: Shouldn't the filter be `${key}:*` instead?

**Review by jonathan-gramain — COMMENTED** [review]
I wonder if we should not apply this fix to 7.4 as well

File: lib/api/BackbeatAPI.js (hunk 2 above)
Comment (bennettbuchanan) [reply lib/api/BackbeatAPI.js:622]: I think either will work for this case, but I agree that keeping the `:` is a bit more readable. I will double check that the timestamp is always appended to confirm.

File: lib/api/BackbeatAPI.js (hunk 1 above)
Comment (bennettbuchanan) [reply lib/api/BackbeatAPI.js:626]: 👍 I agree, that would be more efficient. My one concern is the case of a big upload where there could be hundreds of interval keys to delete. I'll just confirm some performance assumptions around `batch`.

**Comment by bennettbuchanan** [comment]
@jonathan-gramain No need for this in 7.4 unless we backport this CRR monitor feature as well.

File: lib/api/BackbeatAPI.js (hunk 2 above)
Comment (jonathan-gramain) [reply lib/api/BackbeatAPI.js:622]: My concern would be if we included other keys that begin with the target key as prefix (e.g. you want to process "foo" but you also process "foo/bar", I'm not sure this is a possibility but I wanted to raise the point though.

**Comment by bennettbuchanan** [comment]
Seeing as this is an edge case and we're already far along in the RC process for Zenko 1.1., I’m putting this on hold so that https://scality.atlassian.net/browse/ZENKOIO-11 can take precedence.

**Pull request closed without merging**